blog/2020-03-01-CombiningBPMNWithOptaplanner.adoc

```diff
@@ -0,0 +1,611 @@
+move the downloaded file there.
+
+== Creating the Java Interfaces
+
+In our BPMN, we referenced Java classes and services we still need to create.
+In particular:
+
+* We referenced `com.company.Flight` which is used when creating, modifying
+  and getting the seat assignments of a flight.
+
+* We referenced `com.company.Passenger` which is used when we add a passenger
+  to a flight.
+
+* We referenced the operation `addPassengerToFlight` of
+  `com.company.FlightService` when we add a passenger to the flight.
+
+* We referenced the operation `createSeatAssignments` of
+  `com.company.FlightService` when we create the seat assignments.
+
+We'll be using the following domain model for our classes:
+
+image::ClassDiagram.png[]
+
+* A seat has a seat identifier (ex: "3A") and what type it is.
+
+[source,java]
+----
+public class Seat {
+    @PlanningId
+    public String seatIdentifier;
+    public Type type;
+
+    public enum Type {
+        WINDOW, AISLE, OTHER
+    }
+}
+----
+
+* A Passenger has a name and a preferred seat type (Window, Aisle, etc.) and
+  is assigned a seat on the flight.
+
+[source,java]
+----
+@PlanningEntity
+public class Passenger {
+    @PlanningId
+    public String name;
+
+    @PlanningVariable(valueRangeProviderRefs = "seatRange")
+    public Seat seat;
+
+    public Seat.Type preferredSeatType;
+}
+----
+
+* A Flight has an id, a list of passengers on the flight, and the number
+  of rows of columns on the plane.
+
+[source,java]
+----
+@PlanningSolution
+public class Flight {
+
+    public String id;
+
+    @PlanningScore
+    public HardSoftScore score;
+
+    @PlanningEntityCollectionProperty
+    public List<Passenger> passengerList;
+
+    public int rowCount;
+    public int columnCount;
+
+    @ValueRangeProvider(id = "seatRange")
+    private List<Seat> calculateSeatList() {
+        List<Seat> out = new ArrayList<>(rowCount * columnCount);
+        final int MIDDLE_OF_ROW = columnCount / 2;
+        for (int column = 0; column < columnCount; column++) {
+            for (int row = 0; row < rowCount; row++) {
+                Seat seat = new Seat();
+                // 'A' + n = nth letter of the alphabet
+                seat.seatIdentifier = (row + 1) + String.valueOf((char)('A' + column));
+                seat.type = (column == 0 || column == columnCount - 1)? Seat.Type.WINDOW :
+                    (column == MIDDLE_OF_ROW || column == MIDDLE_OF_ROW + 1)? Seat.Type.AISLE :
+                        Seat.Type.OTHER;
+                out.add(seat);
+            }
+        }
+        return out;
+    }
+}
+----
+
+Now that we have our domain model, we can create our services. For the
+`addPassengerToFlight` service, we can simply add the passenger to the
+flight's passenger list. For the `createSeatAssignments` service,
+we can use `SolverManager` to create a `Solver` for us and wait for the
+final best solution.
+
+[source,java]
+----
+@ApplicationScoped
```

Comment:
Where does this annotation come from? Did you mean `org.springframework.web.context.annotation.ApplicationScope`?